plugins/sigma/src/parse.cpp

```diff
@@ -369,12 +366,10 @@ caf::expected<expression> parse_search_id(const data& yaml) {
       };
       // Helper to create an expression from a (transformed) value.
       auto make_predicate_expr = [&](const data& value) -> expression {
-        // Convert strings to patterns if wildcarding is present.
+        // Convert strings to case-insensitive patterns.
         if (auto str = caf::get_if<std::string>(&value))
-          if (!str->empty())
-            if (auto pat = make_pattern(*str); pat && *pat)
-              return predicate{extractor, relational_operator::equal,
-                               data{std::move(**pat)}};
+          if (auto pat = make_pattern(*str))
+            return predicate{extractor, op, data{std::move(*pat)}};
```

Comment:
The previous version was transpiling all wildcard-strings to a predicate with the `equals` operator, but here we're using the previously defined `op` operator so e.g. our `startswith` will create a predicate with `ni <pattern>` instead of `== <pattern>`. I'm not 100% sure if that should be the correct behavior though, more like 80% sure.